sql/scan.go

```diff
@@ -337,7 +337,16 @@ func (n *scanNode) initWhere(where *parser.Where) error {
 	if n.err == nil {
 		// Normalize the expression (this will also evaluate any branches that are
 		// constant).
-		n.filter, n.err = parser.NormalizeAndTypeCheckExpr(n.filter)
+		n.filter, n.err = parser.NormalizeExpr(n.filter)
+	}
+	var whereType parser.Datum
+	if n.err == nil {
+		whereType, n.err = parser.TypeCheckExpr(n.filter)
+	}
+	if n.err == nil {
```

Comment:
Is there a reason you're dropping out of the if-branch where `whereType` is set in order to check it again? I would have thought this should look like:

```
  if n.err == nil {
    var whereType parser.Datum
    if whereType, n.err = parser.TypeCheckExpr(n.fitler); n.err == nil && whereType != parser.DummyBool {
      n.err = ...
    }
  }
```